react-app/src/components/Common/clinkmenu.js

```diff
@@ -11,135 +13,145 @@ const layout = {
 };
 
 // interactions with clinks
-class ClinkMenu extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { loading: false, editVisible: false, shareVisible: false, };
-  }
+const ClinkMenu = () => {
 
-  showEdit = () => {
-    this.setState({
-      editVisible: true,
-    });
+  const currentToken = localStorage.getItem('currentToken');
+  const users = useSelector(state => state.users.users);
+  const isCurrentUserFetched = useSelector(state => state.users.isCurrentUserFetched);
+  const dispatch = useDispatch();
+
+  const [isLoading, setLoading] = useState(false);
+  const [editIsVisible, setEditVisible] = useState(false);
+  const [shareIsVisible, setShareVisible] = useState(false);
+  useEffect(() => {
+    if (isCurrentUserFetched) {
+      dispatch(fetchUsers(currentToken));
+    }
+  }, []);
+
+  const showEdit = () => {
+    setEditVisible(true);
   };
 
-  showShare = () => {
-    this.setState({
-      shareVisible: true,
-    });
+  const showShare = () => {
+    setShareVisible(true);
   };
 
-  onEditFinish = values => {
-    console.log('Success:', values);
-    this.setState({ loading: true });
+  const onEditFinish = values => {
+    setLoading(true);
```

Comment:
Are these just mocks?

Reply:
yeah actually, we've been doing all of them like that and i forgot to make them real LOL

Reply:
Yes it's just a mock, the actual dispatch is coming in a separate PR!